android/src/main/java/com/chargebee/android/ChargebeeReactNativeModule.kt

```diff
@@ -64,19 +66,22 @@ class ChargebeeReactNativeModule internal constructor(context: ReactApplicationC
       CBPurchase.retrieveProducts(it, productIds,
         object : CBCallback.ListProductsCallback<ArrayList<CBProduct>> {
           override fun onSuccess(productDetails: ArrayList<CBProduct>) {
-            CBPurchase.purchaseProduct(
-              productDetails.first(),
-              customerId,
-              object : CBCallback.PurchaseCallback<String> {
+            if(productDetails.size > 0) {
```

Comment:
Should we handle the `else` case too? 🤔

Reply:
Yes. Will be addressing this case in next PR.